scripts/Install-VSCode.ps1

```diff
@@ -133,118 +133,400 @@ param(
     [ValidateSet("Stable", "Insider-System", "Insider-User")]
     [string]$BuildEdition = "Stable",
 
-    [Parameter()]
-    [switch]$User,
-
     [Parameter()]
     [ValidateNotNull()]
     [string[]]$AdditionalExtensions = @(),
 
     [switch]$LaunchWhenDone,
 
-    [switch]$EnableContextMenus
+    [switch]$EnableContextMenus,
+
+    [switch]$WhatIf
 )
 
-if (($PSVersionTable.PSVersion.Major -le 5) -or $IsWindows) {
-    switch ($Architecture) {
-        "64-bit" {
-            if ((Get-CimInstance -ClassName Win32_OperatingSystem).OSArchitecture -eq "64-bit") {
-                $codePath = $env:ProgramFiles
-                $bitVersion = "win32-x64"
+function Test-IsOsX64 {
+    if ($PSVersionTable.PSVersion.Major -lt 6) {
+        return (Get-CimInstance -ClassName Win32_OperatingSystem).OSArchitecture -eq "64-bit"
+    }
+
+    return [System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture -eq [System.Runtime.InteropServices.Architecture]::X64
```

Comment:
You could probably use `using namespace System.Runtime.InteropServices`

Reply:
Is `using namespace` available in PowerShell v3?

Reply:
ugh... that's right.